### ISSUE TITLE
`file_mount` with `mode: COPY` slower than expected on Google Cloud Storage

### ISSUE THREAD

**Issue by sharvil** [skypilot-org/skypilot#3358]
tl;dr: a single large file copy in the same region from GCS --> GCP (`n1-highmem-4`) should be closer to 250 MB/s, but we're observing ~60 MB/s for file mounts with `mode: COPY`. Details below.

Pasting from a conversation on Slack:

Turns out the IO problem I thought I was experiencing was actually an issue with `gsutil`. Turns out `gsutil cp` from GCS --> GCP in the same region tops out at about 60 MB/s (on a single tens-of-gigs file). `gsutil rsync` for the same file runs at about 160 MB/s.

`gcloud storage cp` runs at about 250 MB/s. From what I can tell, `gcloud` replaces `gsutil` and should be used instead. The default Ubuntu 10 image has an old `gcloud` binary from 2022 so it needs to be upgraded first.

All of this was triggered by what felt like slow sync time for a `file_mount` with `mode: COPY`. Since `file_mount` was slower than I expected, I decided to just do the copy by hand in the `setup` section of my SkyPilot yaml and discovered the above behavior. I'll note that `file_mount` with `mode: COPY` has the same performance as `gsutil cp`.
